src/lib/parquet/src/Flow/Parquet/BinaryReader/Bytes.php

```diff
@@ -36,6 +35,10 @@ public function count() : int
     // IteratorAggregate methods
     public function getIterator() : \ArrayIterator
     {
+        if ($this->iterator === null) {
```

Comment:
just this mnemonic is reducing memory consumption by half due to not keeping repeated bytes in memory.